bug: dev conda enviornment not working with sphinx 8.1.0

I was trying to install the dev dependencies following the instruction from the readme, and with the latest release (v8.1.0 released recently) I get an error when trying to do `make dirhtml`

```bash
Running Sphinx v8.1.0
loading translations [en]... done

Extension error:
Could not import extension sphinxcontrib.mermaid (exception: cannot import name 'ExtensionError' from 'sphinx.util' (.../miniforge3/envs/deployment-docs-dev/lib/python3.12/site-packages/sphinx/util/__init__.py))
make: *** [dirhtml] Error 2
```

If I downgrade to `sphinx=8.0.2` the problem is resolved. Not sure exactly what the problem is but a potential temporary solution would be to pin sphinx. Happy to open a PR, if you think this is the right way to do go about it.

--- a/conda/environments/deployment_docs.yml
+++ b/conda/environments/deployment_docs.yml
@@ -10,7 +10,8 @@ dependencies:
   - pydata-sphinx-theme>=0.15.4
   - python=3.12
   - pre-commit>=3.8.0
-  - sphinx>=8.0.2
+  # Upper bound pin on sphinx can be removed once https://github.com/mgaitan/sphinxcontrib-mermaid/issues/160 is resolved
+  - sphinx>=8.0.2,<8.1
   - sphinx-autobuild>=2024.9.19
   - sphinx-copybutton>=0.5.2
   - sphinx-design>=0.6.1

